app/assets/javascripts/tictactoe.js

```diff
@@ -1 +1,166 @@
-// Code your JavaScript / jQuery solution here
+var turn = 0
+var gameId = 0
+const winningCombo = [[0, 1, 2], [3, 4, 5], [6, 7, 8], [0, 3, 6], [1, 4, 7], [2, 5, 8], [0, 4, 8], [2, 4, 6]]
+
+$(document).ready(function() {
+  attachListeners();
+
+  $('#games').on('click', '.previous-game', function () {
+    loadGame(this);
+  })
+});
+
+function attachListeners() {
+  // document.querySelectorAll('td').forEach((td) => {
+  //   td.addEventListener('click', (el) => {
+  //     if (el.target.innerText === "" && !checkWinner()) {
+  //       doTurn(el.target)
+  //     }
+  //   })
+  // })
+
+  $('td').on('click', function() {
+    if (!$.text(this) && !checkWinner()) {
+      doTurn(this);
+    }
+  });
+
+  document.getElementById('save').addEventListener('click', () => saveGame())
+  document.getElementById('previous').addEventListener('click', () => previousGame())
+  document.getElementById('clear').addEventListener('click', () => clearGame())
+}
+
+function player() {
+  return (turn % 2 === 0) ? 'X' : 'O'
+}
+
+function doTurn(square) {
```

Comment:
It may be best to abstract checking for a tie in `doTurn`. That way checkWinner is solely checking for a winner.

```javascript
function doTurn(square) {
  updateState(square);
  turn++;
  if (checkWinner()) {
    saveGame();
    resetBoard();
  } else if (turn === 9) {
    setMessage("Tie game.");
    saveGame();
    resetBoard();
  }
}
```